xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/parser/xhtml/wikimodel/XWikiMacroHandler.java

```diff
@@ -62,6 +70,24 @@ public XWikiMacroHandler(ComponentManager componentManager, XHTMLParser parser)
         this.parser = parser;
     }
 
+    private String getSyntax(TagContext context) throws ComponentLookupException
+    {
+        if (context.getTagStack().getStackParameter(CURRENT_SYNTAX) != null) {
+            return (String) context.getTagStack().popStackParameter(CURRENT_SYNTAX);
+
+            // if the current syntax is not retrieved from the context, we get it from the RenderingContext
```

Comment:
You might want to rewrite a little the first part since "contex" vs "RenderingContext" is a bit confusing. Maybe something like "previous nodes" instead.